src/Geta.Net.Extensions/Generators/QueryStringBuilder.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Linq;
+using System.Web;
+
+namespace Geta.Net.Extensions.Generators
+{
+    /// <summary>
+    ///     Helper class for creating and modifying URL's parameters.
+    /// </summary>
+    public class QueryStringBuilder
+    {
+        private readonly UriBuilder _uriBuilder;
+        private readonly bool _isRelative;
+
+        /// <summary>
+        ///     Instantiates new QueryStringBuilder with provided URL.
+        /// </summary>
+        /// <param name="url">URL for which to build QueryStringBuilder.</param>
+        public QueryStringBuilder(string url)
+        {
+            if (Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out var uri))
+            {
+                if (uri.IsAbsoluteUri)
+                {
+                    _uriBuilder = new UriBuilder(uri);
+                }
+                else
+                {
+                    var baseUri = new Uri("http://dummy.com");
+                    var dummyUri = new Uri(baseUri, uri);
+                    _uriBuilder = new UriBuilder(dummyUri);
+                    _isRelative = true;
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Instantiates new QueryStringBuilder with provided Uri.
+        /// </summary>
+        /// <param name="uri">Uri for which to build QueryStringBuilder.</param>
+        public QueryStringBuilder(Uri uri)
+        {
+            _uriBuilder = new UriBuilder(uri);
```

Comment:
If the passed Uri is relative, we have to create an absolute one.

Reply:
so eventually this will be the reduction to only absolute urls? -> public QueryStringBuilder(string host, string pathAndParams)?

Reply:
No, just implement the same logic as with the string URL version.

Reply:
> No, just implement the same logic as with the string URL version.

It's a bit unclear then, please explain in greater details

Reply:
Ok. There are two constructors you implemented - one takes a string as a parameter and the second one takes `Uri`. In the version where you take a string, you "fix" relative URLs so that UriBuilder doesn't blow up. In the `Uri` version, you don't do it. So make both versions the same.